agent/probe/process.go

```diff
@@ -2964,6 +2908,95 @@ func (p *Probe) PatchContainerProcess(pid int, bEval bool) bool {
 	return false
 }
 
+func isControllerType(role string) bool {
+	return role == "controller" || role == "controller+enforcer+manager" || role == "allinone"
+}
+
+func isManagerType(role string) bool {
+	return role == "manager" || role == "controller+enforcer+manager" || role == "allinone"
+}
+
+// protected by lock
+func (p *Probe) isNVChildProcess(nvRole string, proc *procInternal) bool {
+	var err error
+	if nvRole == "" {
+		return false
+	}
+
+	ppath := proc.ppath
+	path := proc.path
+	pid := proc.pid
+	ppid := proc.ppid
+	bRuncInit := p.isAllowRuncInitCommand("runc", proc.cmds)
+	if ppath == "" {
+		if ppath, err = global.SYS.GetFilePath(ppid); err != nil {
+			log.WithFields(log.Fields{"err": err, "nvRole": nvRole, "ppid": ppid}).Debug()
+		}
+	}
+
+	if path == "" && pid > 0 {
+		if path, err = global.SYS.GetFilePath(pid); err != nil {
+			log.WithFields(log.Fields{"err": err, "nvRole": nvRole, "pid": pid}).Debug()
+		}
+	}
+
+	mLog.WithFields(log.Fields{"path": path, "ppath": ppath, "nvRole": nvRole, "ppid": ppid}).Debug()
+	if isManagerType(nvRole) && ppath == "/usr/bin/bash" && path == "/usr/bin/python3.12" {
+		return true /// only for cli case
+	}
+
+	// backup lookup if we did not catch its parents
+	for i := 0; i < 8; i++ { // lookup for 8 callers
+		if global.RT.IsRuntimeProcess(filepath.Base(ppath), nil) || bRuncInit {
+			mLog.WithFields(log.Fields{"path": path, "ppath": ppath, "nvRole": nvRole, "k8s": p.bKubePlatform, "bRuncInit": bRuncInit}).Debug()
+			switch path {
+			case "/usr/bin/cat": // k8s readiness probe
+				if p.bKubePlatform && isControllerType(nvRole) {
+					if bRuncInit {
+						return true
+					} else if len(proc.cmds) > 1 && proc.cmds[1] == "/tmp/ready" {
+						return true
+					}
```

Comment:
Can you elaborate a little more about this newly added code?  It looks like `cat /tmp/ready /etc/passwd` would bypass the check, so the `else if` check would be a little useless.  Can we double confirm that?

Reply:
It should raise an alert. Basically, it will only allow the health check command. Others will be raised an alert.

Reply:
That's exactly my point.  `cat /etc/passwd` would trigger an alert, but `cat /tmp/ready /etc/passwd` won't.  :-)
Would this work slightly better? 
```suggestion
					if bRuncInit {
						return true
					} else if len(proc.cmds) == 2 && proc.cmds[1] == "/tmp/ready" {
						return true
					}
```

Reply:
Fixed.